great_expectations/checkpoint/checkpoint.py

```diff
@@ -99,6 +106,85 @@ def get_substituted_config(
                     self._substituted_config = substituted_config
                 return substituted_config
 
+    def get_runtime_batch_request(
+        self,
+        substituted_runtime_config: CheckpointConfig,
+        validation_batch_request: Union[dict, BatchRequest, None],
+    ) -> BatchRequest:
+        if substituted_runtime_config.batch_request is None:
+            return (
+                validation_batch_request
+                if isinstance(validation_batch_request, (BatchRequest, type(None)))
+                else BatchRequest(**validation_batch_request)
+            )
+
+        # get batch requests as dicts; since get_json_dict() doesn't include batch_data, re-add
+        base_batch_request_batch_data: Any = (
+            substituted_runtime_config.batch_request.batch_data
+        )
+        base_batch_request_dict: dict = (
+            substituted_runtime_config.batch_request.get_json_dict()
+        )
+        base_batch_request_dict["batch_data"] = base_batch_request_batch_data
+
+        if isinstance(validation_batch_request, BatchRequest):
+            validation_batch_request_batch_data: Any = (
+                validation_batch_request.batch_data
+            )
+            validation_batch_request: dict = validation_batch_request.get_json_dict()
+            validation_batch_request["batch_data"] = validation_batch_request_batch_data
+        elif isinstance(validation_batch_request, None):
+            return BatchRequest(**base_batch_request_dict)
+
+        runtime_batch_request_dict = deepcopy(validation_batch_request)
+        for key, val in runtime_batch_request_dict.items():
+            if val is not None and base_batch_request_dict.get(key) is not None:
+                raise CheckpointError(
+                    f"BatchRequest attribute '{key}' was specified in both validation and top-level CheckpointConfig."
```

Comment:
I was wondering if it is OK to specify some keys at a lower (e.g. runtime) level, but not others.  For example, data_asset_name must not be overridden, but can some other fields be over-ridden per the spec?..

Reply:
I suppose it would be fine for `limit` to be overwritten at runtime. Unsure about other keys. Currently, top-level batch_spec keys are allowed to be overwritten during template updates. Here's what spec says:

batch_request: TBD
- update if the key is new
- maybe error if the key is repeated at runtime
- - definitely okay if the key is repeated in a new checkpoint updating a template checkpoint

Reply:
If the spec says that it is not allowed for the keys to be re-specified via the runtime dictionary, then we are probably OK.